src/univariate/affine.jl

```diff
@@ -0,0 +1,222 @@
+"""
+    AffineDistribution(shift, scale, base_dist)
+
+A shifted and scaled (affinely transformed) version of `base_dist`.
+
+If ``Z`` is a random variable with distribution `base_dist`, then `AffineDistribution` with
+parameters `shift`, `scale`, `base_dist` is the distribution of the random variable
+```math
+X = shift + scale * Z
+```
+
+Note that `scale(x) != x.scale` in general; if `base_dist` already has a scale parameter,
+`scale(x::AffineDistribution)` will return `x.scale * scale(base_dist)`.
+
+If `base_dist` is a discrete distribution, the probability mass function of the transformed 
+distribution is given by
+```math
+P(X = x) = P\\left(Z = \\frac{x-shift}{scale} \\right).
+```
+
+If `base_dist` is a continuous distribution, the probability density function of
+the transformed distribution is given by
+```math
+f(x) = \\frac{1}{scale} base_dist \\! \\left( \\frac{x-shift}{scale} \\right).
+```
+
+If `base_dist` falls under a location-scale family, `affine` will attempt to return a 
+transformed distribution of type `base_dist` if an `affine` method has been implemented. 
+Otherwise, it will fall back on the `AffineDistribution` wrapper type.
+
+```julia
+affine(shift, scale, base_dist)  # location-scale transformed distribution
+params(d)            # Get the parameters, i.e. (shift, scale, base_dist)
+location(d)          # Get the location parameter
+scale(d)             # Get the scale parameter
+```
+"""
+struct AffineDistribution{T<:Real, S<:ValueSupport, D<:UnivariateDistribution{S}} <: UnivariateDistribution{S}
+    shift::T
```

Comment:
Shift and scale should be different types ("think matrix, vector", but 1d). Why do we restrict to 1-d?